packages/vscode-extension/src/dependency/DependencyManager.ts

```diff
@@ -1,440 +1,264 @@
+    if (installed) {
+      return "installed";
+    }
+    return "notInstalled";
   }
 
-  public async checkStorybookInstalled() {
-    const status = checkMinDependencyVersionInstalled(
-      "@storybook/react-native",
-      MIN_STORYBOOK_VERSION_SUPPORTED
-    );
-    const installed = status === "installed";
-
-    this.webview.postMessage({
-      command: "isStorybookInstalled",
-      data: {
-        installed,
-        info: "Whether Storybook is installed.",
-        error: undefined,
-        isOptional: true,
-      },
-    });
-    Logger.debug("Storybook installed:", installed);
-    return installed;
-  }
+  private async podsStatus() {
+    if (await isExpoGoProject()) {
+      // Expo Go projects don't need pods
+      return "installed";
+    }
 
-  public async installPods(
-    appRootFolder: string,
-    forceCleanBuild: boolean,
-    cancelToken: CancelToken
-  ) {
+    if (this.stalePods) {
+      return "notInstalled";
+    }
+
+    const appRootFolder = getAppRootFolder();
     const iosDirPath = getIosSourceDir(appRootFolder);
 
+    Logger.debug(`Check pods in ${iosDirPath}`);
     if (!iosDirPath) {
-      this.webview.postMessage({
-        command: "isPodsInstalled",
-        data: {
-          installed: false,
-          info: "Whether iOS dependencies are installed.",
-          error: "iOS directory does not exist",
-        },
-      });
-      throw new Error(`ios directory was not found inside the workspace.`);
+      return "notInstalled";
     }
 
-    const commandInIosDir = (args: string) => {
-      return command(args, {
-        cwd: iosDirPath,
-        env: {
-          ...getLaunchConfiguration().env,
-          LANG: "en_US.UTF-8",
-        },
-      });
-    };
-
-    try {
-      if (forceCleanBuild) {
-        await cancelToken.adapt(commandInIosDir("pod deintegrate"));
-      }
+    const podfileLockExists = fs.existsSync(path.join(iosDirPath, "Podfile.lock"));
+    const podsDirExists = fs.existsSync(path.join(iosDirPath, "Pods"));
 
-      await cancelToken.adapt(commandInIosDir("pod install"));
-    } catch (e) {
-      Logger.error("Pods not installed", e);
-      this.webview.postMessage({
-        command: "isPodsInstalled",
-        data: {
-          installed: false,
-          info: "Whether iOS dependencies are installed.",
-          error: "Unable to install pods",
-        },
-      });
-      return;
+    if (podfileLockExists && podsDirExists) {
+      return "installed";
     }
-
-    this.stalePods = false;
-
-    this.webview.postMessage({
-      command: "isPodsInstalled",
-      data: {
-        installed: true,
-        info: "Whether iOS dependencies are installed.",
-        error: undefined,
-      },
-    });
-    Logger.debug("Project pods installed");
+    return "notInstalled";
   }
 }
 
-export async function checkIfCLIInstalled(cmd: string, options: Record<string, unknown> = {}) {
+async function isCommandInstalled(cmd: string) {
```

Comment:
I'd  just rename to `testCommand`,`probeCommand` or `checkCommand` as the method doesn't verify whether something is installed but only checks whether the provided command runs sucesfully